common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -23,6 +23,58 @@
 log = logging.getLogger(__name__)
 
 
+def _unenroll_entitlement(course_entitlement, course_run_key):
+    """
+    Internal method to handle the details of Unenrolling a User in a Course Run.
+    """
+    CourseEnrollment.unenroll(course_entitlement.user, course_run_key, skip_refund=True)
+    course_entitlement.set_enrollment(None)
+
+
+@transaction.atomic
+def _process_revoke_and_unenroll_entitlement(course_entitlement, is_refund=False):
+    """
+    Process the revoke of the Course Entitlement and refund if needed
+
+    Arguments:
+        course_entitlement: Course Entitlement Object
+
+        is_refund (bool): True if a refund should be processed
+
+    Exceptions:
+        IntegrityError if there is an issue that should reverse the database changes
+    """
+    if course_entitlement.expired_at is None:
+        course_entitlement.expired_at = timezone.now()
+        log.info(
+            'Set expired_at to [%s] for course entitlement [%s]',
+            course_entitlement.expired_at,
+            course_entitlement.uuid
+        )
+        course_entitlement.save()
+
+    if course_entitlement.enrollment_course_run is not None:
+        course_id = course_entitlement.enrollment_course_run.course_id
+        _unenroll_entitlement(course_entitlement, course_id)
+        log.info(
+            'Unenrolled user [%s] from course run [%s] as part of revocation of course entitlement [%s]',
+            course_entitlement.user.username,
+            course_id,
+            course_entitlement.uuid
+        )
+
+    if is_refund:
+        refund_successful = refund_entitlement(course_entitlement=course_entitlement)
+        if not refund_successful:
+            # This state is achieved in most cases by a failure in the ecommerce service to process the refund.
+            log.warn(
+                'Entitlement Refund failed for Course Entitlement [%s], alert User',
+                str(course_entitlement.uuid)
```

Comment:
Why does this need to be cast to a `str` here, but not in the other log messages? Should we also be casting it in those?

Reply:
I can remove the str here it is done by default in the logger I believe. I will double check.